src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      newState = { ...newState, ...action.extraData };
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const removeKey of action.keysToRemove) {
+        delete newState[removeKey];
+      }
+    }
+
+    if (action.type === 'clear') {
+      newState = {};
+    }
```

Comment:
Use a switch statement instead of multiple if statements when you have a limited amount of conditions to improve code readability and structure. Also, ensure to include a default case for error handling as per checklist items 2 and 3.